src/features/user/Widget/DonationLink/DonationLinkForm.tsx

```diff
@@ -0,0 +1,280 @@
+import { ReactElement, useContext, useState, useEffect } from 'react';
+import { Button, styled, Switch, TextField } from '@mui/material';
+import i18next from '../../../../../i18n';
+import AutoCompleteCountry from '../../../common/InputTypes/AutoCompleteCountryNew';
+import { UserPropsContext } from '../../../common/Layout/UserPropsContext';
+import InlineFormDisplayGroup from '../../../common/Layout/Forms/InlineFormDisplayGroup';
+import supportedLanguages from '../../../../utils/language/supportedLanguages.json';
+import React from 'react';
+import ProjectSelectAutocomplete from '../../BulkCodes/components/ProjectSelectAutocomplete';
+import { TENANT_ID } from '../../../../utils/constants/environment';
+const { useTranslation } = i18next;
+import styles from './DonationLinkForm.module.scss';
+import CopyToClipboard from '../../../common/CopyToClipboard';
+import {
+  MuiAutoComplete,
+  StyledAutoCompleteOption,
+} from '../../../common/InputTypes/MuiAutoComplete';
+import { Project } from '../../../common/types/project';
+import { allCountries } from '../../../../utils/constants/countries';
+import CustomizedSnackbars from './CustomSnackBar';
+
+// TODOO - refactor code for reuse?
+const StyledForm = styled('form')((/* { theme } */) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 24,
+  alignItems: 'flex-start',
+  '& .formButton': {
+    marginTop: 24,
+  },
+  '& .inputContainer': {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: 24,
+    width: '100%',
+  },
+}));
+
+const MuiTextField = styled(TextField)(() => {
+  return {
+    flex: '1',
+    minWidth: 240,
+  };
+});
+
+interface DonationLinkFormProps {
+  projectsList: Project[] | null;
+}
+
+interface LanguageType {
+  langCode: string;
+  languageName: string;
+}
+
+const DonationLinkForm = ({
+  projectsList,
+}: DonationLinkFormProps): ReactElement | null => {
+  const { user } = useContext(UserPropsContext);
+  const [country, setCountry] = useState('auto');
+  const [Languages, setLanguage] = useState<LanguageType>({
+    langCode: 'auto',
+    languageName: 'Automatic Selection',
+  });
+  const [donationUrl, setDonationUrl] = useState<string>('');
+  const { t, ready } = useTranslation(['donationLink', 'donate']);
+  const [localProject, setLocalProject] = useState<Project | null>(null);
+  const [isSupport, setIsSupport] = useState<boolean>(!user.isPrivate);
+  const [isTesting, setIsTesting] = useState<boolean>(false);
+  const [isProjectSelected, setIsProjectSelected] = useState<boolean>(false);
+  const [isArrayUpdated, setIsArrayUpdated] = useState<boolean>(false);
+  const [isLinkUpdated, setIsLinkUpdated] = useState<boolean>(false);
+
+  const handleUrlChange = () => {
+    const link = isTesting
+      ? 'http://paydev.pp.eco/'
```

Comment:
Please update this to use `process.env.NEXT_TESTING_DONATION_URL` over here. Set the same in your env.local file to the current redirect of paydev.pp.eco